### ISSUE TITLE
[CI] Node-20 update

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#122115]
### 🐛 Describe the bug

PyTorch Dev Infra needs to update runners to Linux flavor capable of running node-20 as node-16 and 18 are being deprecated by GitHub, see https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

### Versions

CI

cc @ezyang @gchanan @zou3519 @kadeng @seemethere @pytorch/pytorch-dev-infra

**Comment by malfet** [comment]
I've attempted to do it back in Feb https://github.com/pytorch/pytorch/pull/119403 but almost all workflows failed because node 20 can not be executed on the runner

**Comment by huydhn** [comment]
@jeanschmidt @osalpekar Do we plan to upgrade the base Linux image to something newer as part of ARC?

**Comment by PaliC** [comment]
Ohh it's been updated as per https://github.com/pytorch-labs/pytorch-gha-infra/pull/359/

**Comment by malfet** [comment]
Hmm, doesn't look like changes has been applied so far, see https://github.com/pytorch/pytorch/actions/runs/8348588211/job/22850855154?pr=119403 :
/home/ec2-user/actions-runner/externals/node20/bin/node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /home/ec2-user/actions-runner/externals/node20/bin/node)

**Comment by malfet** [comment]
> Ohh it's been updated as per [pytorch-labs/pytorch-gha-infra#359](https://github.com/pytorch-labs/pytorch-gha-infra/pull/359)

No, it was not, as `amzn2-ami-hvm-2.0.20240306.2-x86_64-ebs` still ships with glibc-2.26, but node-20 needs glibc-2.28:
```
sh-4.2$ /lib64/libc.so.6
GNU C Library (GNU libc) stable release version 2.26, by Roland McGrath et al.
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 7.3.1 20180712 (Red Hat 7.3.1-17).
Available extensions:
        crypt add-on version 2.1 by Michael Glad and others
        GNU Libidn by Simon Josefsson
        Native POSIX Threads Library by Ulrich Drepper et al
        BIND-8.2.3-T5B
libc ABIs: UNIQUE IFUNC
For bug reporting instructions, please see:
<http://www.gnu.org/software/libc/bugs.html>.
sh-4.2$ hostnamectl
   Static hostname: ip-10-0-22-178.ec2.internal
         Icon name: computer-vm
           Chassis: vm
        Machine ID: ec2f346dfb846283a1cdcff9ed9d646d
           Boot ID: e1c822da464d4fd6b6831e2b1ea1ac62
    Virtualization: amazon
  Operating System: Amazon Linux 2
       CPE OS Name: cpe:2.3:o:amazon:amazon_linux:2
            Kernel: Linux 4.14.336-257.562.amzn2.x86_64
      Architecture: x86-64
```

**Comment by malfet** [comment]
I.e. we need to move away from AmazonLinux-2 to AmazonLinux-2023, per https://docs.aws.amazon.com/linux/al2023/ug/compare-with-al2.html#glibc-gcc-and-binutils

**Comment by ZainRizvi** [comment]
Once the 2.5.1 release no longer needs these machines, we should be good to remove the last holdout
